packages/build/src/plugins_core/edge_functions/index.ts

```diff
@@ -30,8 +31,8 @@ const coreStep = async function ({
 }) {
   const { edge_functions: declarations = [] } = netlifyConfig
   const { deno_import_map: userDefinedImportMap } = netlifyConfig.functions['*']
-  const distPath = resolve(buildDir, distDirectory)
-  const internalSrcPath = resolve(buildDir, internalSrcDirectory)
+  const distPath = resolve(buildDir, packagePath || '', distDirectory)
```

Comment:
Not a blocker, but for a follow-up it might be nice to set a default value for `packagePath` once at the top level rather than having to do it all over the place.